src/components/Album/Album.js

```diff
@@ -1,39 +1,87 @@
-import React, { useState, useEffect, useContext } from "react"
-import { useParams } from "react-router-dom"
-import './Album.css'
-import Form from '../Form/Form'
-import { getAlbumsByMasterId } from "../../api"
-import UserContext from "../UserContext/UserContext"
+import React, { useState, useEffect, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { getAlbumDetails } from '../../api';
+import Form from '../Form/Form';
+import './Album.css';
+import UserContext from '../UserContext/UserContext';
 
 function Album() {
-  const {setCurrentUser} = useContext(UserContext)
-  const [album, setAlbum] = useState(null)
-  const [isLoading, setIsLoading] = useState(true)
-  const [modal, setModal] = useState(false)
-  
-  const {id} = useParams()
-  
-  const showModal = () => {
-    modal ? setModal(false) : setModal(true)
-  }
+  const { id } = useParams();
+  const [albumDetails, setAlbumDetails] = useState(null);
+  const [isLoading, setLoading] = useState(true);
+  const [modal, setModal] = useState(false);
+  const {currentUser, setCurrentUser} = useContext(UserContext) 
 
   useEffect(() => {
-    getAlbumsByMasterId(id)
-      .then(res => {
-        setAlbum(res)
-        setIsLoading(false)
+    setLoading(true);
+    getAlbumDetails(id)
+      .then(data => {
+        setAlbumDetails(data);
+        setLoading(false);
       })
-  }, [])
-
-  
-  if(!isLoading) {
-    return (
-      <>
-       <p>{id} hello</p>
-       {!modal && <button onClick={showModal}>Add a journal entry</button>}
-       {modal && <Form {...album} setCurrentUser={setCurrentUser} showModal={showModal}/>}
-      </>
-    )
-  } 
+      .catch(error => {
+        console.error('Error fetching album details:', error);
+        setLoading(false);
```

Comment:
i think you can `setLoading(false)` outside the promise so you only have to have it on a single line

Reply:
.catch(error => {
        console.error('Error fetching album details:', error);
      })
      .finally(() => {
        setLoading(false);
      });
      
      used a finally block to let us know that setLoading(false) will be executed regardless of the promise resolve.